<!-- -->
<launch>

	<arg name="base_control_port" default="/dev/U5"/>   <!--底层can分析仪-->
	<arg name="gps_port" default="/dev/ttyS1"/>		    <!-- GPS-->
	<arg name="stm32_port" default="/dev/U6"/>          <!-- STM32-->
	<arg name="max_speed"  default="30.0"/>
	<arg name="forsight_dis" default="5.0"/>
	<arg name="path_file" default="path.txt"/>
	
<!--底层 -->
	<node name="base_control" pkg="little_ant" type="base_control" output="screen">
		<param name ="obd_can_port_name" value="$(arg base_control_port)"/>
		<param name ="stm32_port_name" value="$(arg stm32_port)"/>
		<param name ="stm32_baudrate" value="115200"/>
		<param name ="max_steering_speed" value="2.0"/> <!--前后帧转角最大差值 -->
	</node>
	
	<node name="purepursuit_node" pkg="path_tracking" type="purepursuit_node" output="screen">
		<param name="path_points_file" value="$(find data)/data/raw/$(arg path_file)"/>
		<param name="speed" value="$(arg max_speed)"/>						<!-- 跟踪速度  -->
		<param name="disThreshold" value="$(arg forsight_dis)"/>
	</node>
	
	
	<node pkg="novatel" type="novatel_node" name="novatel_node"  >

		<param name="port" value="$(arg gps_port)" />
		<param name="baudrate" value="115200" />
		<param name="odom_topic" value="/gps_odom" />
		<!--<param name="log_commands" value="inspvaxb ontime 0.05" /> -->
		<param name="log_commands" value="inspvaxb ontime 0.05; bestutmb ontime 0.05" />
		<!-- <param name="configure_port" value="COM2,9600,RTCM,NONE" /> -->
		<!-- <param name="log_commands" value="BESTUTMB ONTIME 1.0; BESTVELB ONTIME 1.0" /> -->
		<param name="gps_default_logs_period" value="0.0" />
		<param name="span_default_logs_period" value="0.0" />
		
		<param name="inspvax_topic" value = "/gps" />
	</node>
	
	<node name="record_test_data_node" pkg="record_test_data" type="record_test_data_node">
		<param name="file_path" value="$(find data)/data/debug/"/>
		<param name="file_name" value="pure_$(arg path_file)_$(arg forsight_dis)"/>
	</node>
	
	<node  type="decision_making_node" pkg ="decision_making" name="decision_making_node" >  
		<param name ="traffic_mark_topic" value="/traffic_mark" />
	</node>
	

</launch>
